### PR TITLE
Add mypy github action matcher

### PR DESCRIPTION
**Pull request by booxter**

### CODE REVIEW

**Comment by booxter** [comment]
@bjhargrave added the matcher. You can see the result here: https://github.com/instructlab/instructlab/pull/1274/files is it what you'd like to see? Let me know; I will drop the top patch then and then we can merge. Thanks!

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -68,11 +68,15 @@ jobs:
         run: |
           echo "::add-matcher::.github/workflows/matchers/pylint.json"
           tox -e lint
+          echo "::remove-matcher owner=pylint-error::"
```
Comment (booxter) [inline .github/workflows/lint.yml:71]: I'm not sure if we have to remove them across separate steps. Maybe someone could confirm.

Comment (bjhargrave) [reply .github/workflows/lint.yml:71]: I think removing the matcher is safest. It can avoid the previous matcher also matching the errors.

**Comment by bjhargrave** [comment]
> is it what you'd like to see?

Yes, it looks as expected! Thanks for adding.

**Comment by booxter** [comment]
@bjhargrave ack. I removed the breaking patch, this should be ready now.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (booxter) [reply .github/workflows/lint.yml:71]: Ack. I am not sure the state of matchers survive across steps; but I guess it won't make it worse.

**Review by leseb — COMMENTED** [review]
I know you don't mind separating commits so how about one commit for removing the pylint matcher and one for adding mypy matcher? Thanks!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (bjhargrave) [inline .github/workflows/lint.yml:71]: We want mypy to run even when lint fails. But this only removes the pylint matcher when lint succeeds. We probably need to move the remove-matcher to a separate step that runs even when lint fails or into the mypy step.

Comment (booxter) [reply .github/workflows/lint.yml:71]: They do survive as proven here: https://github.com/instructlab/instructlab/pull/1286/files

Will update as you suggested.

**Review by bjhargrave — APPROVED** [review]

**Review by russellb — APPROVED** [review]

**Review by leseb — APPROVED** [review]

**Pull request merged**